client/src/templates/Challenges/classic/classic.css

```diff
@@ -17,53 +17,50 @@
   display: none !important;
 }
 
-.action-row {
+.monaco-editor-tabs {
   padding: 10px;
   border-bottom: 1px solid var(--quaternary-background);
+  display: grid;
+  grid-template-columns: repeat(auto-fill, minmax(8rem, 1fr));
 }
 
-.monaco-editor-tabs [aria-expanded='true'],
-.action-row [aria-expanded='true'] {
+.monaco-editor-tabs [aria-expanded='true'] {
   border-color: var(--secondary-color);
   background-color: var(--secondary-color);
   color: var(--secondary-background);
 }
 
-.tabs-row {
-  display: flex;
-  flex-direction: row;
-  justify-content: space-between;
-}
-
-.tabs-row button,
 .monaco-editor-tabs button {
   padding: 6px 12px;
   border-width: 2px;
   border-style: solid;
 }
 
-.monaco-editor-tabs {
-  display: flex;
-  margin-inline-end: auto;
+.preview-buttons-section {
+  display: inline-flex;
+  grid-column-end: -1;
 }
 
-.monaco-editor-tabs button:not(:first-child) {
-  border-left: none;
-}
-
-.panel-display-tabs button:first-child,
-.tabs-row > button:first-child {
-  margin: 0 10px 0 0;
+.editor-console-button {
+  width: 6rem;
+  margin-inline-start: 1.5rem;
+  grid-column-end: -2;
 }
 
 .restart-step-tab {
   margin: 0 auto;
 }
 
 #mobile-layout {
-  display: flex;
-  flex-direction: column;
-  height: 100%;
+  display: grid;
+  min-height: 100vh;
+  min-height: 100dvh;
+  grid-template-rows: 30px 1fr;
+}
+
+#mobile-layout .tab-content {
+  display: grid;
+  grid-template-rows: 1fr auto;
```

Comment:
this is trippy, the editor stacking is weird because the instruction, preview, console, and upper jaw are in the same layout, while editor is inside `.tab-content` layout, here is the trippy part, the instruction are also inside the `.tab-content` and editor layout, but outside it.

Think of the instruction as reverse portal, I don't know how someone created it without using React, but they have actually make the something most people find impossible.

Unfortunately, it is inside an interactive component that have different sizing, which can create weird bugs if not handled carefully.

Currently, there is a weird bug, and it disables the scrolling when you use your mobile device to scroll, you need to use the mobile device not an emulator.

here is a challenge in the main production